【在 TypeScript 中使用】页面按教程来，修改package.json文件react-app-rewired start --scripts-version react-scripts-ts   启动不了

- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://new-issue.ant.design/](https://new-issue.ant.design/)

### Steps to reproduce
"scripts": {
-   "start": "react-scripts-ts start",
    "start": "react-app-rewired start --scripts-version react-scripts-ts",
-   "build": "react-scripts-ts build",
    "build": "react-app-rewired build --scripts-version react-scripts-ts",
-   "test": "react-scripts-ts test --env=jsdom",
    "test": "react-app-rewired test --env=jsdom --scripts-version react-scripts-ts",
}
修改后yarn  start报错Cannot find module '/Users/XXX/react_demo/antd-demo-ts/node_modules/react-scripts-ts/config/webpack.config'

### What is expected?
解决，并修改教程

### What is actually happening?
报错

| Environment | Info |
|---|---|
| antd | 3.12.1 |
| React | xx |
| System | mac  10.12.6 |
| Browser | xx |


<!-- generated by ant-design-issue-helper. DO NOT REMOVE -->

--- a/docs/react/use-in-typescript.en-US.md
+++ b/docs/react/use-in-typescript.en-US.md
@@ -106,119 +106,95 @@ module.exports = function override(config, env) {
 };
 ```
 
-### Use ts-import-plugin
+### Use babel-plugin-import
 
-[ts-import-plugin](https://github.com/Brooooooklyn/ts-import-plugin) is a TypeScript plugin for importing components on demand ([How does it work?](/docs/react/getting-started#Import-on-Demand)). We are now trying to install it and modify `config-overrides.js`.
+[babel-plugin-import](https://github.com/ant-design/babel-plugin-import) is a babel plugin for importing components on demand ([How does it work?](/docs/react/getting-started#Import-on-Demand)). We are now trying to install it and modify `config-overrides.js`.
 
 ```bash
-$ yarn add ts-import-plugin
+$ yarn add babel-plugin-import
 ```
 
-```js
-/* config-overrides.js */
-const tsImportPluginFactory = require('ts-import-plugin');
-const { getLoader } = require("react-app-rewired");
+```diff
++ const { override, fixBabelImports } = require('customize-cra');
 
-module.exports = function override(config, env) {
-  const tsLoader = getLoader(
-    config.module.rules,
-    rule =>
-      rule.loader &&
-      typeof rule.loader === 'string' &&
-      rule.loader.includes('ts-loader')
-  );
-
-  tsLoader.options = {
-    getCustomTransformers: () => ({
-      before: [ tsImportPluginFactory({
-        libraryDirectory: 'es',
-        libraryName: 'antd',
-        style: 'css',
-      }) ]
-    })
-  };
-
-  return config;
-}
+- module.exports = function override(config, env) {
+-   // do stuff with the webpack config...
+-   return config;
+- };
++ module.exports = override(
++   fixBabelImports('import', {
++     libraryName: 'antd',
++     libraryDirectory: 'es',
++     style: 'css',
++   }),
++ );
 ```
 
 Remove the `@import '~antd/dist/antd.css';` statement added before because `babel-plugin-import` will import styles and import components like below:
 
 ```diff
-import * as React from 'react';
-import { Button } from 'antd';
-import './App.css';
+  // src/App.js
+  import React, { Component } from 'react';
+- import Button from 'antd/lib/button';
++ import { Button } from 'antd';
+  import './App.css';
 
-class App extends React.Component {
-  render() {
-    return (
-      <div className="App">
-        <Button type="primary">Button</Button>
-      </div>
-    );
+  class App extends Component {
+    render() {
+      return (
+        <div className="App">
+          <Button type="primary">Button</Button>
+        </div>
+      );
+    }
   }
-}
 
-export default App;
+  export default App;
 ```
 
 Then reboot with `yarn start` and visit the demo page, you should not find any [warning messages](https://zos.alipayobjects.com/rmsportal/vgcHJRVZFmPjAawwVoXK.png) in the console, which prove that the `import on demand` config is working now. You will find more info about it in [this guide](/docs/react/getting-started#Import-on-Demand).
 
 ### Customize Theme
 
-According to the [Customize Theme documentation](/docs/react/customize-theme), to customize the theme, we need to modify `less` variables with tools such as [less-loader](https://github.com/webpack/less-loader). We can also use [react-app-rewire-less](http://npmjs.com/react-app-rewire-less) to achieve this. Import it and modify `config-overrides.js` like below.
+According to the [Customize Theme documentation](/docs/react/customize-theme), to customize the theme, we need to modify `less` variables with tools such as [less-loader](https://github.com/webpack/less-loader). We can also use [addLessLoader](https://github.com/arackaf/customize-cra#addlessloaderloaderoptions) to achieve this. Import it and modify `config-overrides.js` like below.
 
 ```bash
-$ yarn add react-app-rewire-less --dev
+$ yarn add less less-loader
 ```
 
 ```diff
-  const tsImportPluginFactory = require('ts-import-plugin')
-  const { getLoader } = require("react-app-rewired");
-+ const rewireLess = require('react-app-rewire-less');
+- const { override, fixBabelImports } = require('customize-cra');
++ const { override, fixBabelImports, addLessLoader } = require('customize-cra');
 
-  module.exports = function override(config) {
-    const tsLoader = getLoader(
-      config.module.rules,
-      rule =>
-        rule.loader &&
-        typeof rule.loader === 'string' &&
-        rule.loader.includes('ts-loader')
-    );
-
-    tsLoader.options = {
-      getCustomTransformers: () => ({
-        before: [ tsImportPluginFactory({
-          libraryName: 'antd',
-          libraryDirectory: 'es',
--         style: 'css',
-+         style: true,
-        }) ]
-      })
-    };
-
-+   config = rewireLess.withLoaderOptions({
-+     javascriptEnabled: true,
-+     modifyVars: { "@primary-color": "#1DA57A" },
-+   })(config, env);
-
-    return config;
-  }
+module.exports = override(
+  fixBabelImports('import', {
+    libraryName: 'antd',
+    libraryDirectory: 'es',
+    style: true,
+  }),
++ addLessLoader({
++   javascriptEnabled: true,
++   modifyVars: { '@primary-color': '#1DA57A' },
++ }),
+);
 ```
 
 We use `modifyVars` option of [less-loader](https://github.com/webpack/less-loader#less-options) here, you can see a green button rendered on the page after rebooting the start server.
 
-## Alternative way
+## Alternative ways
 
 You can also follow instructions in [Use in create-react-app](/docs/react/use-with-create-react-app.en-US.md), then use [react-app-rewire-typescript][https://github.com/lwd-technology/react-app-rewire-typescript] to setup the TypeScript development environment by yourself.
 
 And you can use [react-scripts-ts-antd](https://www.npmjs.com/package/react-scripts-ts-antd) which includes ts-import-plugin, react-app-rewired, scss, less and etc.You can create a new project that without any configurations by running just one command.
+
+You could also try [craco](https://github.com/sharegate/craco) and [craco-antd](https://github.com/FormAPI/craco-antd) to customize create-react-app webpack config same as customize-cra does.
+
 ```bash
-$ create-react-app my-project --scripts-version=react-scripts-ts-antd
+$ yarn create react-app my-project --scripts-version=react-scripts-ts-antd
 ```
 
-## FAQ
+## Other tutorials
 
-### error TS2605: JSX element type Xxx is not a constructor function for JSX elements.
-
-Before antd 3, You need setting `allowSyntheticDefaultImports` to `true` in tsconfig.json.
+- [create-react-app Adding TypeScript](https://facebook.github.io/create-react-app/docs/adding-typescript)
+- [Migrating from create-react-app-typescript to Create React App](https://vincenttunru.com/migrate-create-react-app-typescript-to-create-react-app/)
+- [react-scripts-ts-antd](https://github.com/SZzzzz/react-scripts-ts-antd)

--- a/docs/react/use-in-typescript.en-US.md
+++ b/docs/react/use-in-typescript.en-US.md
@@ -14,14 +14,13 @@ Ensure your system has installed latest version of [yarn](https://yarnpkg.com) o
 Create a new project named `antd-demo-ts` using yarn.
 
 ```bash
-$ yarn create react-app antd-demo-ts --scripts-version=react-scripts-ts
+$ yarn create react-app antd-demo-ts --typescript
 ```
 
 If you are using npm (we will use yarn in the following instructions, it's ok to replace yarn with npm)
 
 ```bash
-$ npm install -g create-react-app
-$ create-react-app antd-demo-ts --scripts-version=react-scripts-ts
+$ npx create-react-app antd-demo-ts --typescript
 ```
 
 Then we go inside `antd-demo-ts` and start it.
@@ -42,12 +41,12 @@ $ yarn add antd
 Modify `src/App.tsx`, import Button component from `antd`.
 
 ```jsx
-import { Button } from 'antd';
-import * as React from 'react';
+import React, { Component } from 'react';
+import Button from 'antd/lib/button';
 import './App.css';
 
-class App extends React.Component {
-  public render() {
+class App extends Component {
+  render() {
     return (
       <div className="App">
         <Button type="primary">Button</Button>
@@ -83,19 +82,18 @@ Now we need to customize the default webpack config. We can achieve that by usin
 Import react-app-rewired and modify the `scripts` field in package.json.
 
 ```
-$ yarn add react-app-rewired --dev
+$ yarn add react-app-rewired
 ```
-
 
 ```diff
 /* package.json */
 "scripts": {
--   "start": "react-scripts-ts start",
-+   "start": "react-app-rewired start --scripts-version react-scripts-ts",
--   "build": "react-scripts-ts build",
-+   "build": "react-app-rewired build --scripts-version react-scripts-ts",
--   "test": "react-scripts-ts test --env=jsdom",
-+   "test": "react-app-rewired test --env=jsdom --scripts-version react-scripts-ts",
+-   "start": "react-scripts start",
++   "start": "react-app-rewired start",
+-   "build": "react-scripts build",
++   "build": "react-app-rewired build",
+-   "test": "react-scripts test",
++   "test": "react-app-rewired test",
 }
 ```
 
@@ -113,12 +111,12 @@ module.exports = function override(config, env) {
 [ts-import-plugin](https://github.com/Brooooooklyn/ts-import-plugin) is a TypeScript plugin for importing components on demand ([How does it work?](/docs/react/getting-started#Import-on-Demand)). We are now trying to install it and modify `config-overrides.js`.
 
 ```bash
-$ yarn add ts-import-plugin --dev
+$ yarn add ts-import-plugin
 ```
 
 ```js
 /* config-overrides.js */
-const tsImportPluginFactory = require('ts-import-plugin')
+const tsImportPluginFactory = require('ts-import-plugin');
 const { getLoader } = require("react-app-rewired");
 
 module.exports = function override(config, env) {

--- a/docs/react/use-in-typescript.en-US.md
+++ b/docs/react/use-in-typescript.en-US.md
@@ -181,20 +181,18 @@ module.exports = override(
 
 We use `modifyVars` option of [less-loader](https://github.com/webpack/less-loader#less-options) here, you can see a green button rendered on the page after rebooting the start server.
 
+> You could also try [craco](https://github.com/sharegate/craco) and [craco-antd](https://github.com/FormAPI/craco-antd) to customize create-react-app webpack config same as customize-cra does.
+
 ## Alternative ways
 
-You can also follow instructions in [Use in create-react-app](/docs/react/use-with-create-react-app.en-US.md), then use [react-app-rewire-typescript][https://github.com/lwd-technology/react-app-rewire-typescript] to setup the TypeScript development environment by yourself.
+You can also follow instructions in [Use in create-react-app](/docs/react/use-with-create-react-app.en-US.md), then use  to setup the TypeScript development environment by yourself.
 
-And you can use [react-scripts-ts-antd](https://www.npmjs.com/package/react-scripts-ts-antd) which includes ts-import-plugin, react-app-rewired, scss, less and etc.You can create a new project that without any configurations by running just one command.
+And you can use [react-scripts-ts-antd](https://www.npmjs.com/package/react-scripts-ts-antd) which includes ts-import-plugin, react-app-rewired, scss, less and etc. You can create a new project that without any configurations by running just one command.
 
-You could also try [craco](https://github.com/sharegate/craco) and [craco-antd](https://github.com/FormAPI/craco-antd) to customize create-react-app webpack config same as customize-cra does.
+## Alternative ways
 
-```bash
-$ yarn create react-app my-project --scripts-version=react-scripts-ts-antd
-```
-
-## Other tutorials
-
+- [Create React apps (with Typescript and antd) with no build configuration](https://github.com/SZzzzz/react-scripts-ts-antd)
+- [react-app-rewire-typescript][https://github.com/lwd-technology/react-app-rewire-typescript]
+- [ts-import-plugin](https://github.com/Brooooooklyn/ts-import-plugin)
 - [create-react-app Adding TypeScript](https://facebook.github.io/create-react-app/docs/adding-typescript)
 - [Migrating from create-react-app-typescript to Create React App](https://vincenttunru.com/migrate-create-react-app-typescript-to-create-react-app/)
-- [react-scripts-ts-antd](https://github.com/SZzzzz/react-scripts-ts-antd)

--- a/docs/react/use-in-typescript.zh-CN.md
+++ b/docs/react/use-in-typescript.zh-CN.md
@@ -105,108 +105,77 @@ module.exports = function override(config, env) {
 };
 ```
 
-### 使用 ts-import-plugin
+### 使用 babel-plugin-import
 
-[ts-import-plugin](https://github.com/Brooooooklyn/ts-import-plugin) 是一个用于按需加载组件代码和样式的 TypeScript 插件（[原理](/docs/react/getting-started#按需加载)），现在我们尝试安装它并修改 `config-overrides.js` 文件。
+[babel-plugin-import](https://github.com/ant-design/babel-plugin-import) 是一个用于按需加载组件代码和样式的 babel 插件（[原理](/docs/react/getting-started#按需加载)），现在我们尝试安装它并修改 `config-overrides.js` 文件。
 
 ```bash
-$ yarn add ts-import-plugin
+$ yarn add babel-plugin-import
 ```
 
-```js
-/* config-overrides.js */
-const tsImportPluginFactory = require('ts-import-plugin');
-const { getLoader } = require("react-app-rewired");
+```diff
++ const { override, fixBabelImports } = require('customize-cra');
 
-module.exports = function override(config, env) {
-  const tsLoader = getLoader(
-    config.module.rules,
-    rule =>
-      rule.loader &&
-      typeof rule.loader === 'string' &&
-      rule.loader.includes('ts-loader')
-  );
-
-  tsLoader.options = {
-    getCustomTransformers: () => ({
-      before: [ tsImportPluginFactory({
-        libraryDirectory: 'es',
-        libraryName: 'antd',
-        style: 'css',
-      }) ]
-    })
-  };
-
-  return config;
-}
+- module.exports = function override(config, env) {
+-   // do stuff with the webpack config...
+-   return config;
+- };
++ module.exports = override(
++   fixBabelImports('import', {
++     libraryName: 'antd',
++     libraryDirectory: 'es',
++     style: 'css',
++   }),
++ );
 ```
 
 然后移除前面在 `src/App.css` 里全量添加的 `@import '~antd/dist/antd.css';` 样式代码，并且按下面的格式引入模块。
 
 ```diff
-import * as React from 'react';
-import { Button } from 'antd';
-import './App.css';
+  // src/App.tsx
+  import React, { Component } from 'react';
+- import Button from 'antd/lib/button';
++ import { Button } from 'antd';
+  import './App.css';
 
-class App extends React.Component {
-  render() {
-    return (
-      <div className="App">
-        <Button type="primary">Button</Button>
-      </div>
-    );
+  class App extends Component {
+    render() {
+      return (
+        <div className="App">
+          <Button type="primary">Button</Button>
+        </div>
+      );
+    }
   }
-}
 
-export default App;
+  export default App;
 ```
 
 最后重启 `yarn start` 访问页面，antd 组件的 js 和 css 代码都会按需加载，你在控制台也不会看到这样的[警告信息](https://zos.alipayobjects.com/rmsportal/vgcHJRVZFmPjAawwVoXK.png)。关于按需加载的原理和其他方式可以阅读[这里](/docs/react/getting-started#按需加载)。
 
 ### 自定义主题
 
-按照 [配置主题](/docs/react/customize-theme) 的要求，自定义主题需要用到 less 变量覆盖功能。我们可以引入 react-app-rewire 的 less 插件 [react-app-rewire-less](http://npmjs.com/react-app-rewire-less) 来帮助加载 less 样式，同时修改 `config-overrides.js` 文件。
-
-```bash
-$ yarn add react-app-rewire-less --dev
-```
+按照 [配置主题](/docs/react/customize-theme) 的要求，自定义主题需要用到 less 变量覆盖功能。我们可以引入 `customize-cra` 中提供的 less 相关的函数 [addLessLoader](https://github.com/arackaf/customize-cra#addlessloaderloaderoptions) 来帮助加载 less 样式，同时修改 `config-overrides.js` 文件如下。
 
 ```diff
-  const tsImportPluginFactory = require('ts-import-plugin')
-  const { getLoader } = require("react-app-rewired");
-+ const rewireLess = require('react-app-rewire-less');
+- const { override, fixBabelImports } = require('customize-cra');
++ const { override, fixBabelImports, addLessLoader } = require('customize-cra');
 
-  module.exports = function override(config, env) {
-    const tsLoader = getLoader(
-      config.module.rules,
-      rule =>
-        rule.loader &&
-        typeof rule.loader === 'string' &&
-        rule.loader.includes('ts-loader')
-    );
-
-    tsLoader.options = {
-      getCustomTransformers: () => ({
-        before: [ tsImportPluginFactory({
-          libraryName: 'antd',
-          libraryDirectory: 'es',
--         style: 'css',
-+         style: true,
-        }) ]
-      })
-    };
-
-+   config = rewireLess.withLoaderOptions({
-+     javascriptEnabled: true,
-+     modifyVars: { "@primary-color": "#1DA57A" },
-+   })(config, env);
-
-    return config;
-  }
+module.exports = override(
+  fixBabelImports('import', {
+    libraryName: 'antd',
+    libraryDirectory: 'es',
+    style: true,
+  }),
++ addLessLoader({
++   javascriptEnabled: true,
++   modifyVars: { '@primary-color': '#1DA57A' },
++ }),
+);
 ```
 
 这里利用了 [less-loader](https://github.com/webpack/less-loader#less-options) 的 `modifyVars` 来进行主题配置，
-变量和其他配置方式可以参考 [配置主题](/docs/react/customize-theme) 文档。
+变量和其他配置方式可以参考 [配置主题](/docs/react/customize-theme-cn) 文档。
 
 修改后重启 `yarn start`，如果看到一个绿色的按钮就说明配置成功了。
 
@@ -215,12 +184,9 @@ $ yarn add react-app-rewire-less --dev
 你也可以根据 [在 create-react-app 中使用](/docs/react/use-with-create-react-app.zh-CN.md) 的介绍使用 [react-app-rewire-typescript][https://github.com/lwd-technology/react-app-rewire-typescript] 自己来配置 TypeScript 的开发环境。
 
 此外，还可以选择 [react-scripts-ts-antd](https://www.npmjs.com/package/react-scripts-ts-antd)，其中已经配置好了 ts-import-plugin、react-app-rewired、scss、less 等常用功能。只需一条命令即可生成一个免配置的 TypeScript 项目.
+
 ```bash
-$ create-react-app my-project --scripts-version=react-scripts-ts-antd
+$ yarn create react-app my-project --scripts-version=react-scripts-ts-antd
 ```
 
-## 常见问题
-
-### error TS2605: JSX element type Xxx is not a constructor function for JSX elements.
-
-antd 3 以前的版本需要在 tsconfig.json 的 `compilerOptions` 中配置 `"allowSyntheticDefaultImports": true`。
+对于自定义 webpack 配置，你也可以使用 [craco](https://github.com/sharegate/craco) 和 [craco-antd](https://github.com/FormAPI/craco-antd) 来实现和 customize-cra 一样的修改 create-react-app 配置的功能。

--- a/docs/react/use-in-typescript.zh-CN.md
+++ b/docs/react/use-in-typescript.zh-CN.md
@@ -179,20 +179,12 @@ module.exports = override(
 
 修改后重启 `yarn start`，如果看到一个绿色的按钮就说明配置成功了。
 
+> 对于自定义 webpack 配置，你也可以使用 [craco](https://github.com/sharegate/craco) 和 [craco-antd](https://github.com/FormAPI/craco-antd) 来实现和 customize-cra 一样的修改 create-react-app 配置的功能。
+
 ## 其他方案
 
-你也可以根据 [在 create-react-app 中使用](/docs/react/use-with-create-react-app.zh-CN.md) 的介绍使用 [react-app-rewire-typescript][https://github.com/lwd-technology/react-app-rewire-typescript] 自己来配置 TypeScript 的开发环境。
-
-此外，还可以选择 [react-scripts-ts-antd](https://www.npmjs.com/package/react-scripts-ts-antd)，其中已经配置好了 ts-import-plugin、react-app-rewired、scss、less 等常用功能。只需一条命令即可生成一个免配置的 TypeScript 项目.
-
-```bash
-$ yarn create react-app my-project --scripts-version=react-scripts-ts-antd
-```
-
-对于自定义 webpack 配置，你也可以使用 [craco](https://github.com/sharegate/craco) 和 [craco-antd](https://github.com/FormAPI/craco-antd) 来实现和 customize-cra 一样的修改 create-react-app 配置的功能。
-
-## 社区教程
-
+- [Create React apps (with Typescript and antd) with no build configuration](https://github.com/SZzzzz/react-scripts-ts-antd)
+- [react-app-rewire-typescript][https://github.com/lwd-technology/react-app-rewire-typescript]
+- [ts-import-plugin](https://github.com/Brooooooklyn/ts-import-plugin)
 - [create-react-app Adding TypeScript](https://facebook.github.io/create-react-app/docs/adding-typescript)
 - [Migrating from create-react-app-typescript to Create React App](https://vincenttunru.com/migrate-create-react-app-typescript-to-create-react-app/)
-- [react-scripts-ts-antd](https://github.com/SZzzzz/react-scripts-ts-antd)

--- a/docs/react/use-in-typescript.zh-CN.md
+++ b/docs/react/use-in-typescript.zh-CN.md
@@ -190,3 +190,9 @@ $ yarn create react-app my-project --scripts-version=react-scripts-ts-antd
 ```
 
 对于自定义 webpack 配置，你也可以使用 [craco](https://github.com/sharegate/craco) 和 [craco-antd](https://github.com/FormAPI/craco-antd) 来实现和 customize-cra 一样的修改 create-react-app 配置的功能。
+
+## 社区教程
+
+- [create-react-app Adding TypeScript](https://facebook.github.io/create-react-app/docs/adding-typescript)
+- [Migrating from create-react-app-typescript to Create React App](https://vincenttunru.com/migrate-create-react-app-typescript-to-create-react-app/)
+- [react-scripts-ts-antd](https://github.com/SZzzzz/react-scripts-ts-antd)

--- a/docs/react/use-in-typescript.zh-CN.md
+++ b/docs/react/use-in-typescript.zh-CN.md
@@ -14,14 +14,13 @@ title: 在 TypeScript 中使用
 使用 yarn 创建项目。
 
 ```bash
-$ yarn create react-app antd-demo-ts --scripts-version=react-scripts-ts
+$ yarn create react-app antd-demo-ts --typescript
 ```
 
 如果你使用的是 npm（接下来我们都会用 yarn 作为例子，如果你习惯用 npm 也没问题）。
 
 ```bash
-$ npm install -g create-react-app
-$ create-react-app antd-demo-ts --scripts-version=react-scripts-ts
+$ npx create-react-app antd-demo-ts --typescript
 ```
 
 然后我们进入项目并启动。
@@ -42,11 +41,11 @@ $ yarn add antd
 修改 `src/App.tsx`，引入 antd 的按钮组件。
 
 ```jsx
-import * as React from 'react';
+import React, { Component } from 'react';
 import Button from 'antd/lib/button';
 import './App.css';
 
-class App extends React.Component {
+class App extends Component {
   render() {
     return (
       <div className="App">
@@ -82,19 +81,18 @@ export default App;
 引入 react-app-rewired 并修改 package.json 里的启动配置。
 
 ```
-$ yarn add react-app-rewired --dev
+$ yarn add react-app-rewired
 ```
-
 
 ```diff
 /* package.json */
 "scripts": {
--   "start": "react-scripts-ts start",
-+   "start": "react-app-rewired start --scripts-version react-scripts-ts",
--   "build": "react-scripts-ts build",
-+   "build": "react-app-rewired build --scripts-version react-scripts-ts",
--   "test": "react-scripts-ts test --env=jsdom",
-+   "test": "react-app-rewired test --env=jsdom --scripts-version react-scripts-ts",
+-   "start": "react-scripts start",
++   "start": "react-app-rewired start",
+-   "build": "react-scripts build",
++   "build": "react-app-rewired build",
+-   "test": "react-scripts test",
++   "test": "react-app-rewired test",
 }
 ```
 
@@ -112,12 +110,12 @@ module.exports = function override(config, env) {
 [ts-import-plugin](https://github.com/Brooooooklyn/ts-import-plugin) 是一个用于按需加载组件代码和样式的 TypeScript 插件（[原理](/docs/react/getting-started#按需加载)），现在我们尝试安装它并修改 `config-overrides.js` 文件。
 
 ```bash
-$ yarn add ts-import-plugin --dev
+$ yarn add ts-import-plugin
 ```
 
 ```js
 /* config-overrides.js */
-const tsImportPluginFactory = require('ts-import-plugin')
+const tsImportPluginFactory = require('ts-import-plugin');
 const { getLoader } = require("react-app-rewired");
 
 module.exports = function override(config, env) {

--- a/docs/react/use-with-create-react-app.en-US.md
+++ b/docs/react/use-with-create-react-app.en-US.md
@@ -176,7 +176,7 @@ According to the [Customize Theme documentation](/docs/react/customize-theme), t
 
 ```bash
 $ yarn add less
-$ yarn add --dev less-loader
+$ yarn add less-loader
 ```
 
 ```diff
@@ -197,6 +197,8 @@ module.exports = override(
 ```
 
 We use `modifyVars` option of [less-loader](https://github.com/webpack/less-loader#less-options) here, you can see a green button rendered on the page after rebooting the start server.
+
+> You could also try [craco](https://github.com/sharegate/craco) and [craco-antd](https://github.com/FormAPI/craco-antd) to customize create-react-app webpack config same as customize-cra does.
 
 ## eject
 

--- a/docs/react/use-with-create-react-app.en-US.md
+++ b/docs/react/use-with-create-react-app.en-US.md
@@ -175,8 +175,7 @@ Then reboot with `yarn start` and visit the demo page, you should not find any [
 According to the [Customize Theme documentation](/docs/react/customize-theme), to customize the theme, we need to modify `less` variables with tools such as [less-loader](https://github.com/webpack/less-loader). We can also use [addLessLoader](https://github.com/arackaf/customize-cra#addlessloaderloaderoptions) to achieve this. Import it and modify `config-overrides.js` like below.
 
 ```bash
-$ yarn add less
-$ yarn add less-loader
+$ yarn add less less-loader
 ```
 
 ```diff

--- a/docs/react/use-with-create-react-app.en-US.md
+++ b/docs/react/use-with-create-react-app.en-US.md
@@ -9,16 +9,10 @@ title: Use in create-react-app
 
 ## Install and Initialization
 
-We need to install `create-react-app` first, you may need install [yarn](https://github.com/yarnpkg/yarn/) too.
+Before all start, you may need install [yarn](https://github.com/yarnpkg/yarn/).
 
 ```bash
-$ npm install -g create-react-app yarn
-```
-
-Create a new project named `antd-demo`.
-
-```bash
-$ create-react-app antd-demo
+$ yarn create react-app antd-demo
 ```
 
 The tool will create and initialize environment and dependencies automatically,

--- a/docs/react/use-with-create-react-app.zh-CN.md
+++ b/docs/react/use-with-create-react-app.zh-CN.md
@@ -173,7 +173,7 @@ $ yarn add babel-plugin-import
 
 ```bash
 $ yarn add less
-$ yarn add --dev less-loader
+$ yarn add less-loader
 ```
 
 ```diff
@@ -197,6 +197,8 @@ module.exports = override(
 变量和其他配置方式可以参考 [配置主题](/docs/react/customize-theme) 文档。
 
 修改后重启 `yarn start`，如果看到一个绿色的按钮就说明配置成功了。
+
+> 你也可以使用 [craco](https://github.com/sharegate/craco) 和 [craco-antd](https://github.com/FormAPI/craco-antd) 来实现和 customize-cra 一亿的修改 create-react-app 配置的功能。
 
 ## eject
 

--- a/docs/react/use-with-create-react-app.zh-CN.md
+++ b/docs/react/use-with-create-react-app.zh-CN.md
@@ -9,16 +9,10 @@ title: 在 create-react-app 中使用
 
 ## 安装和初始化
 
-我们需要在命令行中安装 create-react-app 工具，你可能还需要安装 [yarn](https://github.com/yarnpkg/yarn/)。
+在开始之前，你可能需要安装 [yarn](https://github.com/yarnpkg/yarn/)。
 
 ```bash
-$ npm install -g create-react-app yarn
-```
-
-然后新建一个项目。
-
-```bash
-$ create-react-app antd-demo
+$ yarn create react-app antd-demo
 ```
 
 工具会自动初始化一个脚手架并安装 React 项目的各种必要依赖，如果在过程中出现网络问题，请尝试配置代理或使用其他 npm registry。

--- a/docs/react/use-with-create-react-app.zh-CN.md
+++ b/docs/react/use-with-create-react-app.zh-CN.md
@@ -169,11 +169,10 @@ $ yarn add babel-plugin-import
 
 ### 自定义主题
 
-按照 [配置主题](/docs/react/customize-theme) 的要求，自定义主题需要用到 less 变量覆盖功能。我们可以引入 `customize-cra` 中提供的 less 相关的函数 [addLessLoader](https://github.com/arackaf/customize-cra#addlessloaderloaderoptions) 来帮助加载 less 样式，同时修改 `config-overrides.js` 文件。
+按照 [配置主题](/docs/react/customize-theme) 的要求，自定义主题需要用到 less 变量覆盖功能。我们可以引入 `customize-cra` 中提供的 less 相关的函数 [addLessLoader](https://github.com/arackaf/customize-cra#addlessloaderloaderoptions) 来帮助加载 less 样式，同时修改 `config-overrides.js` 文件如下。
 
 ```bash
-$ yarn add less
-$ yarn add less-loader
+$ yarn add less less-loader
 ```
 
 ```diff
@@ -198,7 +197,7 @@ module.exports = override(
 
 修改后重启 `yarn start`，如果看到一个绿色的按钮就说明配置成功了。
 
-> 你也可以使用 [craco](https://github.com/sharegate/craco) 和 [craco-antd](https://github.com/FormAPI/craco-antd) 来实现和 customize-cra 一亿的修改 create-react-app 配置的功能。
+> 你也可以使用 [craco](https://github.com/sharegate/craco) 和 [craco-antd](https://github.com/FormAPI/craco-antd) 来实现和 customize-cra 一样的修改 create-react-app 配置的功能。
 
 ## eject
 

